### ISSUE TITLE
Dir.openFile fails on windows for serial ports

### ISSUE THREAD

**Issue by MasterQ32** [ziglang/zig#5109]
Zig fails to open (existing) serial ports with both NT name (`COM3`) and UNC name (`\\.\COM3`).

Using `COM3` fails with `error.FileNotFound`, using `\\.\COM3` fails with an [`unreachable` in `std/os/windows.zig:162`](https://github.com/ziglang/zig/blob/328eb8ed8dfdc646df28f899267c76e18645da40/lib/std/os/windows.zig#L162)

To reproduce the issue, run the following code:
```zig
const std = @import("std");
pub fn main() !void {
    var serial = try std.fs.cwd().openFile("COM1", .{ .read = true, .write = true });
    defer serial.close();
}
```

Windows documentation for [Communication Devices](https://docs.microsoft.com/en-us/windows/win32/devio/communications-resource-handles) tells that the correct path for serial ports is two two variants used above.

**Comment by MasterQ32** [comment]
Some more information can be found here:
http://www.osronline.com/article.cfm%5Earticle=381.htm

> For example, if a user issues a `CreateFile("COM1",...)` request, the Win32 subsystem modifies the CreateFile request to be a `NtCreateFile("\Global??\COM1",...)` call. This change results in a call to the NtCreateFile handler in kernel mode. The NtCreateFile handler takes the input information and calls the I/O Manager's `IoCreateFile` API, passing in the input name `"\Global??\COM1"` along with about thirteen other parameters. `IoCreateFile` does some basic checking of the input parameters and then passes the input name, thread Security Identifier (SID), and desired access mode to the Object Manager for processing.

Right now, when i pass any of the strings to `openFile()` i get this:

| Input | Result |
|---------|---------|
| `"COM3"` | `error.FileNotFound` |
| `"\\.\COM3"` | *reached unreachable code* |
| `"\GOBAL??\COM3"` | `error.BadPathName` |

**Comment by daurnimator** [comment]
>  "COM3" | error.FileNotFound

This is expected.

> `"\\.\COM3"` 	_reached unreachable code_

which `unreachable` do you hit?

> "\GOBAL??\COM3"
 
Typo? (missing L)

> error.BadPathName

Ah, this is hitting an incorrect validation check in `sliceToPrefixedSuffixedFileW`: 
```zig
            '/', '*', '?', '"', '<', '>', '|' => return error.BadPathName,
```
That needs to be fixed up.

**Comment by MasterQ32** [comment]
> Typo? (missing L)

Yes, but only from transferring from windows to github :D, same error happens when typing the correct `\\GLOBAL??\COM3`

> which unreachable do you hit?

```
PS C:\Users\xq\Desktop\Ashet> .\debugger.exe -P \\.\COM3
reached unreachable code
???:?:?: 0x7ff6affd4e9d in ??? (???)
???:?:?: 0x7ff6affbf559 in ??? (???)
???:?:?: 0x7ff6affc491a in ??? (???)
???:?:?: 0x7ff6affb6e69 in ??? (???)
???:?:?: 0x7ff6affb1b83 in ??? (???)
???:?:?: 0x7ffadc707bd3 in ??? (???)
???:?:?: 0x7ffadca0ce50 in ??? (???)
```
I don't know from this version of the vbuild, the PDB files are there, but it seems like zig cannot find them when cross-compiling... But it's the `unreachable` in the first post:
```zig
.OBJECT_NAME_INVALID => unreachable,
```
